recipes/LibriSpeech/ASR/CTC/train_with_wav2vec_k2.py

```diff
@@ -0,0 +1,688 @@
+                                         input_lens=wav_lens, 
+                                         graph_compiler=self.graph_compiler,
+                                         texts=texts,
+                                         is_training=is_training)
+
+        loss = loss_ctc
+
+        if stage == sb.Stage.VALID:
+            # Decode token terms to words
+            predicted_texts = self.graph_compiler.decode(
+                p_ctc,
+                wav_lens,
+                ac_scale=self.hparams.ac_scale,
+                decoding_method="1best"
+            ) # list of strings
+            predicted_words = [wrd.split(" ") for wrd in predicted_texts]
+            target_words = [wrd.split(" ") for wrd in texts]
+            self.wer_metric.append(ids, predicted_words, target_words)
+            self.cer_metric.append(ids, predicted_words, target_words)
+        if stage == sb.Stage.TEST:  # Language model decoding only used for test
+            if self.hparams.use_language_modelling:
+                raise NotImplementedError(
+                    "Language modelling is not implemented for models trained with k2"
+                )
+            else:
+                decoding_method=getattr(asr_brain.hparams, "decoding_method", "1best")
+                # If the decoding method is 1best then the metric stats will be
+                # saved in a single file, otherwise, a new directory will be created
+                # for each lm_scale used in whole lattice rescoring.
+                decode_output: Union[dict, List[str]] = self.graph_compiler.decode(
+                    p_ctc,
+                    wav_lens,
+                    search_beam=self.hparams.test_search_beam,
+                    output_beam=self.hparams.test_output_beam,
+                    ac_scale=self.hparams.ac_scale,
+                    max_active_states=self.hparams.test_max_active_state,
+                    is_test=True,
+                    decoding_method=decoding_method,
+                    lm_scale_list=self.hparams.lm_scale_list,
+                ) # list of strings
+                target_words: List[List[str]] = [wrd.split(" ") for wrd in texts]
+                if decoding_method == "1best":
+                    predicted_words: List[List[str]] = [wrd.split(" ") for wrd in decode_output]
+                    self.wer_metric.append(ids, predicted_words, target_words)
+                    self.cer_metric.append(ids, predicted_words, target_words)
+                else:
+                    for i, lm_scale in enumerate(self.hparams.lm_scale_list):
+                        predicted_texts: List[str] = decode_output[f"lm_scale_{lm_scale:.1f}"]
+                        predicted_words: List[List[str]] = [wrd.split(" ") for wrd in predicted_texts]
+                        self.wer_metric[i].append(ids, predicted_words, target_words)
+                        self.cer_metric[i].append(ids, predicted_words, target_words)
```

Comment:
hmmmmmm this should be discussed a bit more. This is quite unusual. Usually, we are only giving the 1-best WER/CER scores.

Reply:
In my use-case, it was pretty handy having the WER stats for each lm scale since I could use that to see how the quality of the LM affects the subs/dels/ins distribution. I personally think that this is a good thing to have, but I can remove it if you disagree.

Reply:
Yep, It would be better to remove it. As I said in another comment, with Orion we can easily search for the best LM scale